ark/type/src/parser/objectLiteral.ts

```diff
@@ -1,85 +1,109 @@
-import type { Dict, error, evaluate, mutable } from "@arktype/util"
-import { node } from "../nodes/parse.js"
-import { PredicateNode } from "../nodes/predicate/predicate.js"
-import { DomainNode } from "../nodes/primitive/domain.js"
-import type { NamedPropsInput } from "../nodes/prop/props.js"
-import { PropsNode } from "../nodes/prop/props.js"
-import { TypeNode } from "../nodes/type.js"
 import type { ParseContext } from "../scope.js"
 import type { inferDefinition, validateDefinition } from "./definition.js"
 import type { validateString } from "./semantic/validate.js"
-import { Scanner } from "./string/shift/scanner.js"
+import {
+	IndexedKey,
+	OptionalValue,
+	parseKeyValueEntry,
+	parsedEntry,
+	validateObjectValue
+} from "./shared.js"
+import type { parseEntry } from "./shared.js"
+import {
+	type Dict,
+	type error,
+	type evaluate,
+	type mutable
+} from "@arktype/util"
 
 export const parseObjectLiteral = (def: Dict, ctx: ParseContext) => {
 	const named: mutable<NamedPropsInput> = {}
+
```

Comment:
Here's how I'd split this up:

```ts
const stringAndSymbolicEntriesOf = (o: Record<string | symbol, unknown>) => [...Object.entries(o), ...Object.getOwnPropertySymbols(o).map((k) => [k, o[k]] as const)]

export const parseObjectLiteral = (def: Dict, ctx: ParseContext) => {
	const named: mutable<NamedPropsInput> = {}
    for(const entry of stringAndSymbolicEntriesOf(def)) {
        // rename parseKeyValueEntry to this (entry implies key value)
        // have it accept entry directly instead of k/v as individual params
        const { innerKey, innerValue, optional } = parseEntry(entry)
	    ctx.path.push(innerKey as string)
        // rename "definition" to valueNode since this is the parsed Node, not a definition
	    const valueNode = ctx.scope.parse(innerValue, ctx)
	    named[innerKey] = {
		    prerequisite: false,
		    value: valueNode,
		    optional
	    }
	    ctx.path.pop()
    }
}
```